### ISSUE TITLE
Exception during refreshing tree item of NOT connected cluster

### ISSUE THREAD

**Issue by odockal** [redhat-developer/intellij-kubernetes#108]
It might be better for user to show some error/warning dialog/notification instead of just intellij's thrown exception like this.

```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://api.openshift4-crs.crs.codereadyqe.com:6443/api/v1/namespaces. Message: Forbidden! User kube:admin/api-openshift4-crs-crs-codereadyqe-com:6443 doesn't have permission. namespaces is forbidden: User "system:anonymous" cannot list resource "namespaces" in API group "" at the cluster scope.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:589)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:526)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:492)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:451)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:433)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:166)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:640)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:79)
	at org.jboss.tools.intellij.kubernetes.model.resource.NonNamespacedResourcesProvider.loadAllResources(NonNamespacedResourcesProvider.kt:36)
	at org.jboss.tools.intellij.kubernetes.model.resource.NonNamespacedResourcesProvider.getAllResources(NonNamespacedResourcesProvider.kt:28)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.getAllResources(ActiveContext.kt:132)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.access$getAllResources(ActiveContext.kt:58)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext$namespacedProviders$2.invoke(ActiveContext.kt:72)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext$namespacedProviders$2.invoke(ActiveContext.kt:58)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.getNamespacedProviders(ActiveContext.kt)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.invalidate(ActiveContext.kt:318)
	at org.jboss.tools.intellij.kubernetes.model.ResourceModel.invalidate(ResourceModel.kt:133)
	at org.jboss.tools.intellij.kubernetes.model.ResourceModel.invalidate(ResourceModel.kt:119)
	at org.jboss.tools.intellij.kubernetes.tree.TreeStructure$Descriptor.invalidate(TreeStructure.kt:248)
	at org.jboss.tools.intellij.kubernetes.actions.RefreshAction.actionPerformed(RefreshAction.kt:20)
	at com.redhat.devtools.intellij.common.actions.TreeAction.actionPerformed(TreeAction.java:101)
	at com.redhat.devtools.intellij.common.actions.TreeAction.actionPerformed(TreeAction.java:95)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:258)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:275)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:287)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:283)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:106)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:277)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:120)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:111)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:522)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:35)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:544)
	at java.awt.Component.processMouseEvent(Component.java:6548)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4526)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:664)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by adietish** [comment]
@odockal: Is this duplicate to #112?

**Comment by adietish** [comment]
@odockal: what do you mean by "NOT connected cluster"? Is this a cluster that's listed but not the active one? Is it a cluster that's the active one but you're not authenticated (cluster freshly started but oc login not executed yet)?

**Comment by adietish** [comment]
The stacktrace shows that the cluster is running but the user is not authenticated. This is the same issue as in #124. In #124 fixes for #112 are already taken into account which results in slightly different behaviour. #124 is thus more accurate to track the issue.
Closing as duplicate to #124.
